frontend/console/src/features/command-pallete/CommandPalette.tsx

```diff
@@ -0,0 +1,98 @@
+import { Combobox, ComboboxInput, ComboboxOption, ComboboxOptions, Dialog, DialogBackdrop, DialogPanel } from '@headlessui/react'
+import { ArrowRight01Icon, CellsIcon } from 'hugeicons-react'
+import { useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { useSchema } from '../../api/schema/use-schema'
+import { type PaletteItem, paletteItems } from './command-palette.utils'
+
+type CommandPaletteProps = {
+  isOpen: boolean
+  onClose: () => void
+}
+
+export const CommandPalette: React.FC<CommandPaletteProps> = ({ isOpen, onClose }) => {
+  const navigate = useNavigate()
+  const { data: schemaData } = useSchema()
+  const [query, setQuery] = useState('')
+  const [items, setItems] = useState<PaletteItem[]>([])
+
+  useEffect(() => {
+    if (schemaData) {
+      const newItems = paletteItems(schemaData)
+      setItems(newItems)
+    }
+  }, [schemaData])
+
+  const filteredItems = query === '' ? [] : items.filter((item) => item.title.toLowerCase().includes(query.toLowerCase()))
+
+  const handleClose = () => {
+    onClose()
+    setQuery('')
+  }
+
+  if (!isOpen) return
+
+  return (
+    <Dialog className='relative z-10' open={isOpen} onClose={handleClose}>
+      <DialogBackdrop
+        transition
+        className='fixed inset-0 bg-gray-900 dark:bg-gray-900 bg-opacity-40 dark:bg-opacity-60 transition-opacity data-[closed]:opacity-0 data-[enter]:duration-300 data-[leave]:duration-200 data-[enter]:ease-out data-[leave]:ease-in'
+      />
+
+      <div className='fixed inset-0 z-10 w-screen overflow-y-auto p-4 sm:p-6 md:p-20'>
+        <DialogPanel
+          transition
+          className='mx-auto max-w-xl transform rounded-xl bg-white dark:bg-gray-800 p-2 shadow-2xl ring-1 ring-black dark:ring-gray-700 ring-opacity-5 transition-all data-[closed]:scale-95 data-[closed]:opacity-0 data-[enter]:duration-300 data-[leave]:duration-200 data-[enter]:ease-out data-[leave]:ease-in'
```

Comment:
`max-w-xl` sounds so aggressive relative to what that class actually does hahaha